environment_switcher/lib/src/environment_switcher.dart

```diff
@@ -0,0 +1,195 @@
+import 'package:flutter/material.dart';
+
+import 'environment.dart';
+import 'environment_store.dart';
+import 'select_environment_sheet.dart';
+
+export 'environment.dart';
+export 'environment_store.dart';
+
+class EnvironmentSwitcher extends StatefulWidget {
+  final Widget Function(Environment) builder;
+  final List<Environment> environments;
+  final Environment defaultEnvironment;
+  final EnvironmentStore environmentStore;
+  final bool showBanner;
+
+  /// A banner that visually shows the user what [Environment]
+  /// is currently set to.
+  /// This is useful because it's possible to be on one Flavour
+  /// in the native code, and another in the Flutter code.
```

Comment:
This comment is actually wrong now. We're not setting different icons/app titles ect for environments, so they're entirely flutter-side.